### ISSUE TITLE
Improve checking of font types and fallbacks in CSS

### ISSUE THREAD

**Issue by tofi86** [IDPF/epubcheck#586]
## OpenType fonts with TrueType outlines (*.ttf) shouldn't generate a WARNING

```
WARNING (CSS-007) at "Dummy_EPUB3.epub/OEBPS/stylesheet.css" (line 18, col 71):
   Font-face reference OEBPS/fonts/PT-Sans-Italic.ttf refers to non-standard font type application/x-font-ttf.
```

This WARNING is not true.
The `PT-Sans-Italic.ttf` is an OpenType font with TrueType outlines which should be perfectly valid.
However, it's declared as `application/x-font-ttf` in the `content.opf` file which may seem wrong, but it's still the correct mime-type for OpenType with TrueType outlines.

Also, epubcheck seems to ignore the `format()` property of the `@font-face` declaration:
```css
@font-face{
    font-family: 'PT-Sans';
    font-style: italic;
    font-weight: normal;
    src: url("fonts/PT-Sans-Italic.ttf") format('opentype');
}
```
Seems as epubcheck only checks the file extension and throws a warning for `*.ttf` in general instead of checking the `format()` property and then examining the font file itself.

**Comment by rdeltour** [comment]
Right. I've been reported this privately too.

> it's declared as application/x-font-ttf in the content.opf file which may seem wrong, but it's still the correct mime-type for OpenType with TrueType outlines.

That's the root of EpubCheck's warning: the Publications 3.0.1 spec only lists `application/vnd.ms-opentype` as a [Core Media Type](http://www.idpf.org/epub/301/spec/epub-publications.html#sec-core-media-types).

> Seems as epubcheck only checks the file extension and throws a warning for *.ttf in general 

Right. I'm not sure since when exactly, but it seems this wasn't the case in v3.0.1. This is too restrictive anyway, given that the standard [CSS font selection rules](http://www.w3.org/TR/css3-fonts/#font-style-matching) is an acceptable fallback mechanism.

We should:
- accept `*.ttf` if declared with the right media type
- only report if the font family is actually used in a `font-family` property with the set of rules applied to a valid CSS selector, and when no other fallback font family is referenced (even generic ones like e.g. `sans-serif`). Even then, a RS is expected to render given the default system font so I'm not sure it s/b a warning really.

**Comment by tofi86** [comment]
Sound good, thanks Romain! :+1:

**Comment by jacqueslelezard** [comment]
Hi, is there a way to avoid any warning/info message when validating an epub with epubcheck 4.0.1 using .ttf fonts? For now .ttt font still generate an INFO message.

**Comment by mhenes** [comment]
Any update ^^^ thanks.

**Comment by tofi86** [comment]
@mhenes when you read the linked issue https://github.com/IDPF/epub-revision/issues/667 you will see that you should use `application/font-sfnt` in the future instead of `application/x-font-ttf`.

So I assume there won't be any changes to the epubcheck library to suppress the current INFO message.
